Sources/ServiceDiscovery/ServiceDiscovery.swift

```diff
@@ -61,9 +61,58 @@ public protocol ServiceDiscovery: AnyObject {
     ///                 including cancellation requested through `CancellationToken`.
     ///
     /// -  Returns: A `CancellationToken` instance that can be used to cancel the subscription in the future.
-    func subscribe(to service: Service, onNext nextResultHandler: @escaping (Result<[Instance], Error>) -> Void, onComplete completionHandler: @escaping (CompletionReason) -> Void) -> CancellationToken
+    func subscribe(to service: Service,
+                   onNext nextResultHandler: @escaping (Result<[Instance], Error>) -> Void,
+                   onComplete completionHandler: @escaping (CompletionReason) -> Void) -> CancellationToken
 }
 
+// MARK: - Async service discovery protocol
+
+#if compiler(>=5.5)
+/// Provides service instances lookup.
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public protocol AsyncServiceDiscovery: AnyObject {
```

Comment:
pains me to have to call this `AsyncServiceDiscovery`... I almost prefer we replace the existing `ServiceDiscovery` with the new concurrency constructs and have version 2.0.0 be a completely breaking changes - but with a nicer API

Reply:
Swift >= 5.5 only?

Reply:
@tomerd @ktoso How about https://github.com/apple/swift-service-discovery/pull/35/commits/d36b27d9e8ce285c100203c7c1637e4f6216f361?

The code is not tested and is placed in a separate module for now to avoid compilation errors.

Reply:
That looks good but the async stream isn't fully stable yet I think so perhaps let's wait out a bit?